phoenix-core/src/main/java/org/apache/phoenix/schema/types/PBlob.java

```diff
@@ -17,15 +17,50 @@
  */
 package org.apache.phoenix.schema.types;
 
+import org.apache.phoenix.schema.SortOrder;
+
 import java.io.InputStream;
+import java.sql.Blob;
 import java.sql.Types;
 
 public class PBlob extends PVarbinary {
 
-    public static final PBlob INSTANCE = new PBlob();
+    public static final PBlob INSTANCE = new PBlob(new FileSystemBasedLobStore());
+
+    private final LobStore store;
 
-    private PBlob() {
+    private PBlob(LobStore store) {
         super("BLOB", Types.BLOB, InputStream.class, null, 70);
```

Comment:
I couldn't find the "standard" ordinal for BLOB, so I just used 48 and that works without hacking the array index stuff.